File: core/src/main/java/com/xmartlabs/base/core/log/analytics/TrackableAnalytic.java
package com.xmartlabs.base.core.log.analytics;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.util.Map;

/** Trackable analytic that represents something to be tracked. For example: an event, a screen, others... */
public interface TrackableAnalytic<T> {
  /** @return string the trackable analytic name */
  @NonNull
  String getName();

  /** @return {@code Map<String, ?>} parameters that this trackable analytic should track */
  @Nullable
  Map<String, ?> getParameters();

  /** @return {@code <T>} Analityc type. The type of analytic to track, such as Event, Screen or other type */
  @NonNull
  T getAnaliticType();
}

File: core/src/main/java/com/xmartlabs/bigbang/core/log/analytics/TrackableAnalytic.kt
package com.xmartlabs.bigbang.core.log.analytics

/** Trackable analytic that represents something to be tracked. For example: an event, a screen, others...  */
interface TrackableAnalytic<out T> {
  /** @return string the trackable analytic name
   */
  val name: String

  /** @return `Map<String, Any>` parameters that this trackable analytic should track
   */
  val parameters: Map<String, Any>?

  /** @return `<T>` Analityc type. The type of analytic to track, such as Event, Screen or other type
   */
  val analiticType: T
}
